packages/manager/modules/pci/src/projects/project/kubernetes/details/node-pool/add/add.controller.js

```diff
@@ -61,12 +62,33 @@ export default class {
         ),
       )
       .catch((error) => {
-        this.CucCloudMessage.error(
-          this.$translate.instant('kube_add_node_pool_error', {
-            nodePoolName: this.nodePool.name,
-            message: get(error, 'data.message', error.message),
-          }),
-        );
+        if (get(error, 'data.status') === 412) {
+          // If error code is 412
+          const errorMessage = get(error, 'data.message');
+          const errorId = errorMessage.slice(
+            errorMessage.indexOf('[') + 1,
+            errorMessage.indexOf(']'),
+          );
+          const quotaUrl = this.coreURLBuilder.buildURL(
+            'public-cloud',
+            `#/pci/projects/${this.projectId}/quota`,
+          );
+          const translateMessage = this.$translate.instant(
+            `kube_add_node_pool_error_${errorId}`,
+          );
+          this.CucCloudMessage.error({
+            textHtml: `${translateMessage} <a class="oui-link_icon" href="${quotaUrl}">${this.$translate.instant(
+              'kube_add_node_pool_error_quota_link',
+            )} <span class="oui-icon oui-icon-external-link" aria-hidden="true"></span></a>`,
+          });
+        } else {
+          this.CucCloudMessage.error(
+            this.$translate.instant('kube_add_node_pool_error', {
+              nodePoolName: this.nodePool.name,
+              message: get(error, 'data.message', error.message),
+            }),
+          );
+        }
```

Comment:
Looks like this logic is duplicated, maybe it could be extracted so we avoid duplication